### ISSUE TITLE
load_dataset_builder should error if "name" does not exist?

### ISSUE THREAD

**Issue by severo** [huggingface/datasets#3011]
```
import datasets as ds
builder = ds.load_dataset_builder('sent_comp', name="doesnotexist")
builder.info.config_name
```

returns

```
'doesnotexist'
```

Shouldn't it raise an error instead?

**Comment by lhoestq** [comment]
Yes I think it should raise an error. Currently it looks like it instantiates a custom configuration with the name given by the user:
https://github.com/huggingface/datasets/blob/ba27ce33bf568374cf23a07669fdd875b5718bc2/src/datasets/builder.py#L391-L397

**Comment by ebarkhordar** [comment]
This raises now.

```python
>>> load_dataset_builder("lhoestq/demo1", name="doesnotexist")
ValueError: BuilderConfig 'doesnotexist' not found. Available: ['default']
```

Same on a local data dir, and on a dataset that really has configs (`nyu-mll/glue` lists all twelve in the message). The check in `_create_builder_config` was always there, but it is guarded by `and self.BUILDER_CONFIGS`, and that used to be empty for a plain data-files dataset. Now that configs are built from the README metadata every hub or local dataset carries at least `default`, so the guard fires.

Checked against the releases: 2.14.0 still does what you describe, `info.config_name` comes back as `'doesnotexist'`. 2.16.1 and everything after it raises. Measured on main at 3e2c1a6c3.

`sent_comp` went away with the script datasets so I used demo1, which has the single `default` config your example relies on.
